src/Adapter/Customer/QueryHandler/GetCustomerInformationHandler.php

```diff
@@ -0,0 +1,590 @@
+        );
+    }
+
+    /**
+     * @param Customer $customer
+     *
+     * @return PersonalInformation
+     */
+    private function getPersonalInformation(Customer $customer)
+    {
+        $customerStats = $customer->getStats();
+
+        $gender = new Gender($customer->id_gender, $this->contextLangId);
+        $socialTitle = $gender->name ?: $this->translator->trans('Unknown', [], 'Admin.Orderscustomers.Feature');
+
+        if ($customer->birthday && '0000-00-00' !== $customer->birthday) {
+            $birthday = sprintf(
+                $this->translator->trans('%1$d years old (birth date: %2$s)', [], 'Admin.Orderscustomers.Feature'),
+                $customerStats['age'],
+                Tools::displayDate($customer->birthday)
+            );
+        } else {
+            $birthday = $this->translator->trans('Unknown', [], 'Admin.Orderscustomers.Feature');
+        }
+
+        $registrationDate = Tools::displayDate($customer->date_add, null, true);
+        $lastUpdateDate = Tools::displayDate($customer->date_upd, null, true);
+        $lastVisitDate = $customerStats['last_visit'] ?
+            Tools::displayDate($customerStats['last_visit'], null, true) :
+            $this->translator->trans('Never', [], 'Admin.Global')
+        ;
+
+        $customerShop = new Shop($customer->id_shop);
+        $customerLanguage = new Language($customer->id_lang);
+
+        $customerSubscriptions = new Subscriptions(
+            (bool) $customer->newsletter,
+            (bool) $customer->optin
+        );
+
+        return new PersonalInformation(
+            $customer->firstname,
+            $customer->lastname,
+            $customer->email,
+            $customer->isGuest(),
+            $socialTitle,
+            $birthday,
+            $registrationDate,
+            $lastUpdateDate,
+            $lastVisitDate,
+            $this->getCustomerRankBySales($customer->id),
+            $customerShop->name,
+            $customerLanguage->name,
+            $customerSubscriptions,
+            (bool) $customer->active
+        );
+    }
+
+    /**
+     * @param int $customerId
+     *
+     * @return int|null customer rank or null if customer is not ranked
+     */
+    private function getCustomerRankBySales($customerId)
+    {
+        $sql = 'SELECT SUM(total_paid_real) FROM ' . _DB_PREFIX_ . 'orders WHERE id_customer = ' . (int) $customerId . ' AND valid = 1';
+
+        if ($totalPaid = Db::getInstance()->getValue($sql)) {
+            $sql = '
+                SELECT SQL_CALC_FOUND_ROWS COUNT(*)
+                FROM ' . _DB_PREFIX_ . 'orders
+                WHERE valid = 1
+                    AND id_customer != ' . (int) $customerId . '
+                GROUP BY id_customer
+                HAVING SUM(total_paid_real) > ' . (int) $totalPaid
+            ;
+
+            Db::getInstance()->getValue($sql);
+
+            return (int) Db::getInstance()->getValue('SELECT FOUND_ROWS()') + 1;
```

Comment:
And you use `FOUND_ROWS` after `SQL_CALC_FOUND_ROWS` and `COUNT(*)` 🤔 . I'm sorry I don't get it.
Even if this comes from legacy code, we need to improve it.

Reply:
🆙